### PR TITLE
Expand environment variables in the learning transport connection string

### PR DESCRIPTION
**Pull request by ramonsmits**
This make it possible to do for example `%TEMP%\.learningTransport`

### CODE REVIEW

**Comment by danielmarbach** [comment]
This will only properly work if you do it also for `LearningTransportCustomization` connection strings where we set the storage directory. Pushed an update

**Comment by danielmarbach** [comment]
I actually think this would be a great feature enhancement to Core @ramonsmits Maybe send in a to the core learning transport?

**Comment by ramonsmits** [comment]
@danielmarbach I considered that first but as learning transport is initialized via code in core and not via a connection string (I think..) adding ENVVAR expanding is up to the developer.

File: src/ServiceControl.Transports.Learning/LearningTransportCustomization.cs [hunk 1]
```diff
@@ -49,14 +50,14 @@ public override IProvideQueueLength CreateQueueLengthProvider()
         static void CustomizeEndpoint(EndpointConfiguration endpointConfig, TransportSettings transportSettings, TransportTransactionMode transportTransactionMode)
         {
             var transport = endpointConfig.UseTransport<LearningTransport>();
-            transport.StorageDirectory(transportSettings.ConnectionString);
+            transport.StorageDirectory(Environment.ExpandEnvironmentVariables(transportSettings.ConnectionString));
```
Comment (danielmarbach) [inline src/ServiceControl.Transports.Learning/LearningTransportCustomization.cs:53]: With this we can even support it via the config

**Pull request merged**